### PR TITLE
Allow numbers and boolean values for tags.

### LINKED ISSUE #828: Allow setting numbers as tags
Having the ability to set custom number values which are indexed in Elasticsearch allows to create aggregated graphs based on transactions.

For example, you could track the number of search results for a search transaction, along with the search query to get a list of the top searches without results.

### PR DESCRIPTION
**Pull request by simitt**
Store numbers as scaled_floats, scaling_factor 1000000
fixes #828.

depends on https://github.com/elastic/beats/pull/9772

### CODE REVIEW

**Comment by felixbarny** [comment]
What is the maximum number that can be stored with this scaled float format? Are strings, numbers and booleans stored in the same place in ES? What if one span has the tag `foo: bar` and another `foo: 42` (different data types for the same tag)?

**Comment by simitt** [comment]
Yes, strings, numbers and booleans are stored at the same place in ES. If the same attribute has different datatypes this will throw an exception.

**Comment by graphaelli** [comment]
jenkins, retest this please

**Comment by simitt** [comment]
@felixbarny I'd like to merge this soonish - do you have any concerns regarding the handling?

**Comment by felixbarny** [comment]
Do users have to possibility to set another mapping type in case they want to set large numbers?

For most use cases, including recording bytes up to 9PB and Unix timestamps in up to millisecond precision the current range is sufficient.

**Comment by felixbarny** [comment]
If the user can't configure the mapping type I think it would make sense to validate the range in the APM Server so that the error appears in the agent logs as well rather than only in the APM Server and Elasticsearch logs.

But if possible I think users should be able to configure the mapping type, although the default you choose sounds good to me.

**Comment by simitt** [comment]
There is a way to add fields to the ES template, see discussions around this https://github.com/elastic/apm-server/issues/1371, which makes use of the experimental feature  [append_fields](https://www.elastic.co/guide/en/beats/filebeat/6.x/configuration-template.html) (experimental). 
This can **not** be used for tags, as it raises an exception when used for fields that already have a template. 

However, one can make use of the `setup.template.path` and `setup.template.overwrite: true` config options to point to a custom `fields.yml`, and add other types [here](https://github.com/elastic/apm-server/pull/1712/files#diff-2d1b3f471cc1866ddb1f08e68b28cb0cR36). The custom fields.yml then needs to be manually updated every time the server is updated, to ensure new mappings are added, so all features  work as expected. 

So to summarize, there is some hackish way to do add custom mapping types.

**Comment by simitt** [comment]
jenkins, retest this please.

**Comment by simitt** [comment]
> If the user can't configure the mapping type I think it would make sense to validate the range in the APM Server so that the error appears in the agent logs as well rather than only in the APM Server and Elasticsearch logs.

That is not possible with the setup we have at the moment, as the server would need to either handle the whole request syncronously so errors can be returned back to the agents, or the server would need to fetch all indexed `tags` with their types and check them for every incoming document, which would require some kind of shared cache to work between multiple servers. 

The same handling applies if e.g. a mapping explosion or any other issue with storing the data to ES happens.
